api/v1/configurationpolicy_types.go

```diff
@@ -148,61 +119,54 @@ func (e EvaluationInterval) parseInterval(interval string) (time.Duration, error
-	Severity          Severity          `json:"severity,omitempty"` // low, medium, high
-	RemediationAction RemediationAction `json:"remediationAction"`  // enforce, inform
-	// 'namespaceSelector' defines the list of namespaces to include/exclude for objects defined in
-	// spec.objectTemplates. All selector rules are ANDed. If 'include' is not provided but
-	// 'matchLabels' and/or 'matchExpressions' are, 'include' will behave as if ['*'] were given. If
-	// 'matchExpressions' and 'matchLabels' are both not provided, 'include' must be provided to
-	// retrieve namespaces.
-	NamespaceSelector Target `json:"namespaceSelector,omitempty"`
-	// 'object-templates' and 'object-templates-raw' are arrays of objects for the configuration
-	// policy to check, create, modify, or delete on the cluster. 'object-templates' is an array
-	// of objects, while 'object-templates-raw' is a string containing an array of objects in
-	// YAML format. Only one of the two object-templates variables can be set in a given
-	// configurationPolicy.
-	ObjectTemplates []*ObjectTemplate `json:"object-templates,omitempty"`
-	// 'object-templates' and 'object-templates-raw' are arrays of objects for the configuration
-	// policy to check, create, modify, or delete on the cluster. 'object-templates' is an array
-	// of objects, while 'object-templates-raw' is a string containing an array of objects in
-	// YAML format. Only one of the two object-templates variables can be set in a given
-	// configurationPolicy.
-	ObjectTemplatesRaw string             `json:"object-templates-raw,omitempty"`
-	EvaluationInterval EvaluationInterval `json:"evaluationInterval,omitempty"`
-	// +kubebuilder:default:=None
-	PruneObjectBehavior PruneObjectBehavior `json:"pruneObjectBehavior,omitempty"`
-}
+// ComplianceType describes how objects on the cluster should be compared with the configuration
+// policy's object definition. Options are "MustHave", "MustOnlyHave", or "MustNotHave".
+//
+// +kubebuilder:validation:Enum=MustHave;Musthave;musthave;MustOnlyHave;Mustonlyhave;mustonlyhave;MustNotHave;Mustnothave;mustnothave
+type ComplianceType string
 
-// ObjectTemplate describes how an object should look
-type ObjectTemplate struct {
-	// ComplianceType specifies whether it is: musthave, mustnothave, mustonlyhave
-	ComplianceType ComplianceType `json:"complianceType"`
+const (
+	// MustNotHave is a ComplianceType to not match an object definition.
+	MustNotHave ComplianceType = "MustNotHave"
 
-	MetadataComplianceType MetadataComplianceType `json:"metadataComplianceType,omitempty"`
+	// MustHave is a ComplianceType to match an object definition as a subset of the whole object.
+	MustHave ComplianceType = "MustHave"
 
-	// ObjectDefinition defines required fields for the object
-	// +kubebuilder:pruning:PreserveUnknownFields
-	ObjectDefinition runtime.RawExtension `json:"objectDefinition"`
+	// MustOnlyHave is a ComplianceType to match an object definition exactly with the object.
```

Comment:
I am sure what mustonlyHave does but the previous "MustOnlyHave is an enforcement state to exclusively include a resource" and this one doesn't match(I feel). exclusive vs exact

Reply:
This particular text doesn't wind up in the CRD, so I'm not sure the wording is entirely important here (I thought about removing it... 😬 ). 

Personally I think my wording is more accurate than "exclusively".

Reply:
Good, just curious, Can I use `musthaveonly` in case that name or namespace is not specified?

Reply:
No, the same rules apply where the config-policy-controller needs to have a name to handle an object. The namespace could be supplied by the `namespaceSelector`, though.